python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -52,33 +52,42 @@
     from pyspark.pandas.series import Series  # noqa: F401 (SPARK-34943)
 
 
-def is_valid_operand_for_numeric_arithmetic(operand: Any, *, allow_bool: bool = True) -> bool:
-    """Check whether the operand is valid for arithmetic operations against numerics."""
+def is_valid_operand_for_numeric_arithmetic(
+    operand: Any, *, allow_bool_index_ops: bool = True, allow_bool: bool = True
```

Comment:
Out of curiosity, why do we need `allow_bool_index_ops`? Is there a case where only bool index should be allowed alone?

Reply:
There is a case `allow_bool_index_ops` is not allowed, thus, `allow_bool` is allowed alone.

`allow_bool_index_ops` is set `True` for arithmetics operations against numeric Series and Indexes in order to enable cases as
```py
>>> ps.Series([1, 2, 3]) + ps.Series([True, False, True])
0    2                                                                          
1    2
2    4
dtype: int64
```

However, the case below is not supported now.
```py
>>> ps.Series([True, False, True]) + ps.Series([True, False, True])
Traceback (most recent call last):
...
TypeError: Addition can not be applied to booleans and the given type.
```

The two cases above both use `is_valid_operand_for_numeric_arithmetic` as an input validation, so `allow_bool_index_ops` might be needed.

How do you think about that?

Reply:
Good question. There is NOT a case where only bool index should be allowed alone, but there is a case `allow_bool_index_ops` is `False`, whereas `allow_bool` is `True`, for example
```py
>>> ps.Series([True, False, True]) + ps.Series([True, False, True])
Traceback (most recent call last):
...
TypeError: Addition can not be applied to booleans and the given type.

>>> ps.Series([True, False, True]) + True
0    True
1    True
2    True
dtype: bool
```

In case you are interested, there is a case `allow_bool_index_ops` and `allow_bool` are both `True`, for example
```py
>>> ps.Series([1, 2, 3]) + ps.Series([True, False, True])
0    2
1    2
2    4
dtype: int64
>>> ps.Series([1, 2, 3]) + True
0    2
1    3
2    4
dtype: int64
```

And there is a case `allow_bool_index_ops` and `allow_bool` are both `False`, for example
```py
>> ps.Series([True, False, True]) - ps.Series([True, False, True])
Traceback (most recent call last):
...
TypeError: Subtraction can not be applied to booleans and the given type.
>>> ps.Series([True, False, True]) - True
Traceback (most recent call last):
...
TypeError: Subtraction can not be applied to booleans and the given type.

```

That's why `allow_bool_index_ops` might be needed. Does that make sense?

Reply:
For the first example, pandas works for both cases:

```py
>>> pd.__version__
'1.2.4'

>>> pd.Series([True, False, True]) + pd.Series([True, False, True])
0     True
1    False
2     True
dtype: bool

>>> pd.Series([True, False, True]) + True
0    True
1    True
2    True
dtype: bool
```

so `allow_bool_index_ops` and `allow_bool` should be `True` for this case as well. Then we can consolidate the two?

Reply:
Makes sense! https://issues.apache.org/jira/browse/SPARK-35681 is created for supporting arithmetic operators (+, *) among bool Series/Index, the parameters will be consolidated at that time.

Reply:
BTW, I investigated Nan None True evaluation before at https://github.com/apache/spark/pull/24234/files#diff-2086df5f0cf7291ee71bbde83662c0e34600c53c04ee0e377ceadf61e6c496deR365-R382. might be helpful

Reply:
The arithmetic operators (+, *) among bool Series/Index has been implemented in this PR; tickets have been adjusted.
`allow_bool_index_ops` parameter has been removed.

Reply:
Thanks @HyukjinKwon, that's helpful!